tests/transaction_tests.rs

```diff
@@ -0,0 +1,401 @@
+
+    if HashSet::from_iter(
+        vec!["Byzantium", "Constantinople", "EIP158"]
+            .into_iter()
+            .map(String::from)
+            .collect::<Vec<String>>(),
+    ).is_subset(&expect.network)
+    {
+        // Since Spurious Dragon
+        assert!(network_id.is_some() || network_id.unwrap() == 1u32.into());
+    } else {
+        // Before Spurious Dragon
+        assert!(network_id.is_none());
+    }
+}
+
+/// Takes a path to JSON file and returns a test
+fn make_test(path: PathBuf) -> Option<Vec<TestDescAndFn>> {
+    // For now all the test and filler data is parsed upfront,
+    // to only create tests that contains data that we're able to parse.
+    // This means only tests that have filler "transaction" values can be verified.
+    // Once serde-rlp's decoder is merged upstream we can do two way verification.
+
+    // Test case is always an object with a single key
+    // Grab name of the actual test together with its value
+
+    let fixtures = load_fixtures(path.as_path());
+    assert_eq!(fixtures.len(), 1);
+    let (_, fixtures) = fixtures.into_iter().nth(0).unwrap();
+    // Load filler data
+    let filler = load_filler(&fixtures);
+    assert_eq!(filler.len(), 1);
+    let (_name, filler) = filler.into_iter().nth(0).unwrap();
+
+    // Obvious expected failure as there are no expect values
+    if filler.expect.len() == 0 {
+        let mut desc = TestDesc::new(DynTestName(format!(
+            "{}",
+            path.strip_prefix(get_fixtures_path())
+                .unwrap()
+                .to_string_lossy()
+                .to_string()
+        )));
+        assert!(filler.transaction.is_none());
+        desc.should_panic = ShouldPanic::Yes;
+
+        let test = TestDescAndFn {
+            desc: desc,
+            testfn: DynTestFn(Box::new(move || {
+                test_fn(&fixtures, &filler, None);
+            })),
+        };
+
+        return Some(vec![test]);
+    }
+
+    // This stores all tests per all networks
+    let mut tests = Vec::new();
+
+    for expect in &filler.expect {
+        // let networks = vec!["a", "b"].fjoin(",");
+        let networks = expect
+            .network
+            .iter()
+            .map(|s| &s[..])
+            .collect::<Vec<&str>>()
+            .join(",");
+        // for network in expect.network.iter() {
+        let mut desc = TestDesc::new(DynTestName(format!(
+            "{}@{}@{}",
+            path.strip_prefix(get_fixtures_path())
+                .unwrap()
+                .to_string_lossy()
+                .to_string(),
+            networks,
+            expect.result
+        )));
+
+        desc.should_panic = if &expect.result == "invalid" {
+            ShouldPanic::Yes
+        } else if &expect.result == "valid" {
+            ShouldPanic::No
+        } else {
+            panic!("Unknown expect result {}", &expect.result);
+        };
+
+        // TODO: I couldn't figure better way to do it
```

Comment:
maybe expand on what you are trying to do here?

Reply:
Updated. I was trying to get around borrow checker somehow, but cloning seemed to be simpliest solution. I'll give it a try again. It might decrease the tests time as some ETH tests are 10MB+